py/dml/codegen.py

```diff
@@ -1445,63 +1444,61 @@ def convert_decl(decl_ast):
 
     decls = list(map(convert_decl, decls))
 
-    def mk_sym(name, typ):
-        sym = LocalSymbol(name, scope.unique_cname(name), type=typ,
-                          site=stmt.site, stmt=True)
-        stmts.append(sym_declaration(sym))
-        return sym
-
-    def add_var(name, typ, init = None):
-        sym = scope.add_variable(
-            name, type = typ, site = stmt.site, init = init, stmt = True,
-            make_unique=not dml.globals.debuggable)
-        stmts.append(sym_declaration(sym))
-        return sym
+    def mk_sym(name, typ, mkunique=not dml.globals.debuggable):
+        cname = scope.unique_cname(name) if mkunique else name
+        return LocalSymbol(name, cname, type=typ, site=stmt.site, stmt=True)
 
     if method_call_init:
-        late_sym_adds = []
-        late_declares = []
-        tgts = []
+        syms_to_add = []
+        tgt_syms = []
+        late_declared_syms = []
 
         for (name, typ) in decls:
+            sym = mk_sym(name, typ)
             tgt_typ = safe_realtype_shallow(typ)
             if tgt_typ.const:
                 nonconst_typ = tgt_typ.clone()
                 nonconst_typ.const = False
-                sym = mk_sym('_tmp_' + name, nonconst_typ)
-                late_declares.append((name, typ, sym))
+                tgt_sym = mk_sym('_tmp_' + name, nonconst_typ, True)
+                sym.init = ExpressionInitializer(mkLocalVariable(stmt.site,
+                                                                 tgt_sym))
+                late_declared_syms.append(sym)
             else:
-                sym = mk_sym(name, typ)
-                late_sym_adds.append(sym)
-            tgts.append(mkLocalVariable(stmt.site, sym))
+                tgt_sym = sym
```

Comment:
I suppose the `const` adventures come from `local const int x = m()` where `m` throws?

Would be nice with a short reminder (just a comment with an example is enough) to remind the reader of this use case. (not really new for this PR so not blocking)

Reply:
Yeah, and also to deal with multiple return values
```C
local (const int x, const int y) = m();
```

But yeah, that's old logic. What this change is (mostly) about is so that:
```C
method m() -> (int) { ... }
...
local int x = m();
```
Compiles to:
```C
int x = m();
```
rather than
```C
int x;
x = m();
```

Reply:
what's the point of that change? `int x = m()` looks better to a human, but I don't think gcc cares. Unless the variable is `const`, of course.

Reply:
Before the patch where I allowed method calls as initializers to local variable declarations, then it was genned as:
```
int x = m();
```
so the current behavior can be seen as a regression, which has always bugged me, in particular because it makes dealing with constness more awkward. So this tweak to the logic is to address that regression. It's just a straight-up improvement.